File: 3206-alternating-groups-i/3206-alternating-groups-i.py
class Solution:
    def numberOfAlternatingGroups(self, colors: List[int]) -> int:
        size,count = len(colors),0
        for i in range(size):
            if colors[i%size]!=colors[(i+1)%size] and colors[(i+2)%size]!=colors[(i+1)%size]:
                count+=1
        return count

File: 3206-alternating-groups-i/3206-alternating-groups-i.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors) {
        int count = 0;
        for(int i=0;i<colors.size();i++){
            if(colors[i%colors.size()]!=colors[(i+1)%colors.size()] && colors[(i+2)%colors.size()]!=colors[(i+1)%colors.size()])
                count++;
        }
        return count;
    }
};